exercises/forth/test/Tests.hs

```diff
@@ -1,74 +1,78 @@
 {-# LANGUAGE OverloadedStrings #-}
-import Test.HUnit (Assertion, (@=?), runTestTT, Test(..), Counts(..))
-import System.Exit (ExitCode(..), exitWith)
-import Forth (ForthError(..), evalText, empty, formatStack)
-import Control.Monad (foldM)
-import Data.Text (Text)
 
-exitProperly :: IO Counts -> IO ()
-exitProperly m = do
-  counts <- m
-  exitWith $ if failures counts /= 0 || errors counts /= 0 then ExitFailure 1 else ExitSuccess
+import Control.Monad     (foldM)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
 
-testCase :: String -> Assertion -> Test
-testCase label assertion = TestLabel label (TestCase assertion)
+import Forth (ForthError(..), empty, evalText, formatStack)
 
 main :: IO ()
-main = exitProperly $ runTestTT $ TestList
-       [ TestList forthTests ]
-
-runTexts :: [Text] -> Either ForthError Text
-runTexts xs = formatStack <$> foldM (flip evalText) empty xs
-
-forthTests :: [Test]
-forthTests =
-  [ testCase "no input, no stack" $
-    "" @=? formatStack empty
-  , testCase "numbers just get pushed onto the stack" $
-    Right "1 2 3 4 5" @=? runTexts ["1 2 3 4 5"]
-  , testCase "non-word characters are separators" $
-    -- Note the Ogham Space Mark ( ), this is a spacing character.
```

Comment:
>  think my main question is - now that this line appears in the README, when someone looks at that, what do you think their reaction will be? Will it be "I'll immediately search for all instances of this   character to see what this is all about", or will it be "Why are you telling me this? I will forget about it". Well, it's not my place to predict how people will act.

Maybe it doesn't belong in README...

After some thought, I think that we should simply remove the comment. It is too specific!

If it is part of testing non-word character, it doesn't deserve a comment more than NULL, SOH or tabs. If it is really a special case, it should have its own test.

What do you think?